src/software/fastAnalogWrite.cpp

```diff
@@ -30,7 +30,7 @@ analogInfo analogSetup(PinName pin, uint32_t freq, uint32_t startingValue) {
 
 // Sets the value to output on the analog pin using the structure that was returned when setting it up
 // Value should be in range of 0 and 4095 (PWM_MAX_DUTY_CYCLE)
-void analogSet(analogInfo* pinInfo, uint32_t value) {
+void analogSet(analogInfo* pinInfo, uint16_t value) {
```

Comment:
I see what you did here, but the timer compare function needs a uint32_t, so it doesn't matter anyway.

Reply:
Yes, you are right. STM32 is a 32-bit MCU.